modules/contact/test/tests/mortar_tm/2d/ad_frictional/finite_stiff.i

```diff
@@ -0,0 +1,231 @@
+    elem_type = ${elem}
+    boundary_name_prefix = plank
+  []
+  [plank_id]
+    type = SubdomainIDGenerator
+    input = plank
+    subdomain_id = 1
+  []
+
+  [block]
+    type = GeneratedMeshGenerator
+    dim = 2
+    xmin = 0.31
+    xmax = 0.91
+    ymin = 7.7
+    ymax = 8.5
+    nx = 3
+    ny = 4
+    elem_type = ${elem}
+    boundary_name_prefix = block
+    boundary_id_offset = 10
+  []
+  [block_id]
+    type = SubdomainIDGenerator
+    input = block
+    subdomain_id = 2
+  []
+
+  [combined]
+    type = MeshCollectionGenerator
+    inputs = 'plank_id block_id'
+  []
+  [block_rename]
+    type = RenameBlockGenerator
+    input = combined
+    old_block = '1 2'
+    new_block = 'plank block'
+  []
+[]
+
+[GlobalParams]
+  displacements = 'disp_x disp_y'
+[]
+
+[Variables]
+  [disp_x]
+    order = ${order}
+    block = 'plank block'
+    scaling = '${fparse 2.0 / (E_plank + E_block)}'
+  []
+  [disp_y]
+    order = ${order}
+    block = 'plank block'
+    scaling = '${fparse 2.0 / (E_plank + E_block)}'
+  []
+[]
+
+[Modules/TensorMechanics/Master]
+  [action]
+    strain = FINITE
+    generate_output = 'stress_xx stress_yy stress_zz vonmises_stress hydrostatic_stress strain_xx '
+                      'strain_yy strain_zz'
+    block = 'plank block'
+    use_automatic_differentiation = true
+  []
+[]
+
+[Contact]
+  [frictional]
+    primary = plank_right
+    secondary = block_left
+    formulation = mortar
+    model = coulomb
+    c_normal = 1e8
+    c_tangential = 1e8
+    friction_coefficient = 0.1
+    #normal_lm_scaling = 1.0e-5
+    #tangential_lm_scaling = 1.0e-5
+  []
+[]
+
+[BCs]
+  [left_x]
+    type = ADDirichletBC
+    variable = disp_x
+    boundary = plank_left
+    value = 0.0
+    preset = false
+  []
+  [left_y]
+    type = ADDirichletBC
+    variable = disp_y
+    boundary = plank_bottom
+    value = 0.0
+    preset = false
+  []
+  [right_x]
+    type = ADFunctionDirichletBC
+    variable = disp_x
+    boundary = block_right
+    function = '-0.04*sin(4*(t+1.5))+0.02'
+    preset = false
+  []
+  [right_y]
+    type = ADFunctionDirichletBC
+    variable = disp_y
+    boundary = block_right
+    function = '-t'
+    preset = false
+  []
+[]
+
+[Materials]
+  [plank]
+    type = ADComputeIsotropicElasticityTensor
+    block = 'plank'
+    poissons_ratio = 0.3
+    youngs_modulus = ${E_plank}
+  []
+  [block]
+    type = ADComputeIsotropicElasticityTensor
+    block = 'block'
+    poissons_ratio = 0.3
+    youngs_modulus = ${E_block}
+  []
+  [stress]
+    type = ADComputeFiniteStrainElasticStress
+    block = 'plank block'
+  []
+[]
+
+[Executioner]
+  type = Transient
+  solve_type = 'NEWTON'
+  petsc_options = '-snes_converged_reason -ksp_converged_reason'
+  petsc_options_iname = '-pc_type -pc_factor_shift_type -pc_factor_shift_amount'
+  petsc_options_value = 'lu        NONZERO               1e-24'
+  end_time = 5.3 # 13.5
+  dt = 0.12
+  dtmin = 0.12
+  line_search = 'none'
+  nl_div_tol = 1e+100
```

Comment:
```suggestion
  nl_div_tol = 1e100
```
Was this actually necessary? I forget what the default is

Reply:
When I converted the tests, I removed this option only to realize it was necessary. Atom autocomplete tells me the default value is `1e10`. It's relatively easy to go beyond that with contact constraints.